### PR TITLE
feat: Implement Consumer.offsetsForTimes

### PR DESCRIPTION
**Pull request by tmdiep**
Retrieves cursors for timestamps from the `TopicStatsClient` to implement `PubsubLiteConsumer.offsetsForTimes`.

### CODE REVIEW

**Comment by tmdiep** [comment]
Delaying merge until the RPC is publicly visible.

**Comment by tmdiep** [comment]
ComputeTimeCursor is now publicly visible.

File: src/main/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumer.java [hunk 1]
```diff
@@ -466,8 +467,33 @@ public OffsetAndMetadata committed(TopicPartition topicPartition, Duration durat
...
+                  Collectors.toMap(
+                      entry -> entry.getKey(),
+                      entry ->
+                          topicStatsClient.computeCursorForEventTime(
+                              topicPath,
+                              checkTopicGetPartition(entry.getKey()),
+                              Timestamps.fromMillis(entry.getValue()))));
+      ApiFutures.allAsList(cursors.values()).get(duration.toMillis(), TimeUnit.MILLISECONDS);
+
+      // As per KafkaConsumer.offsetsForTimes, null (represented by no map entry) is returned if
```
Comment (dpcollins-google) [inline src/main/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumer.java:483]: this is not the correct behavior. What the api should return is a mapping from the topic partition to the literal null.

File: src/test/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumerTest.java [hunk 2]
```diff
@@ -481,6 +478,22 @@ public void endOffsets() {
     assertThat(output).isEqualTo(ImmutableMap.of(partition2, 22L, partition4, 44L));
   }
 
+  @Test
+  public void offsetsForTimes() {
+    TopicPartition partition2 = new TopicPartition(example(TopicPath.class).toString(), 2);
+    TopicPartition partition4 = new TopicPartition(example(TopicPath.class).toString(), 4);
+    when(topicStatsClient.computeCursorForEventTime(
+            example(TopicPath.class), Partition.of(2), Timestamps.fromMillis(2000)))
+        .thenReturn(
```
Comment (dpcollins-google) [inline src/test/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumerTest.java:487]: please test the nullability behavior of this

File: src/main/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumer.java (hunk 1 above)
Comment (tmdiep) [reply src/main/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumer.java]: Fixed. Now returns a HashMap, which permits null values.

File: src/test/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumerTest.java (hunk 2 above)
Comment (tmdiep) [reply src/test/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumerTest.java:488]: `partition4` below tests the null case.

**Review by dpcollins-google — APPROVED** [review]

**Pull request merged**